src/pages/patch/PatchRestartModal.tsx

```diff
@@ -28,28 +30,119 @@ import { queryString } from "utils";
   visible,
   onOk,
   onCancel,
   patchId: patchIdFromProps,
   refetchQueries,
   childPatches,
+  setShouldAbortInProgressTasks,
+  shouldAbortInProgressTasks,
+  shouldRestart,
+  setMutationLoading,
+  setButtonDisabled,
+  buttonDisabled,
 }) => {
-  const dispatchToast = useToastContext();
   const { id } = useParams<{ id: string }>();
   const patchId = patchIdFromProps ?? id;
-  const [shouldAbortInProgressTasks, setShouldAbortInProgressTasks] = useState(
-    false
+  return (
+    <>
+      <PatchRestartTasks
+        patchId={patchId}
+        visible={visible}
+        onOk={onOk}
+        onCancel={onCancel}
+        refetchQueries={refetchQueries}
+        setShouldAbortInProgressTasks={setShouldAbortInProgressTasks}
+        shouldAbortInProgressTasks={shouldAbortInProgressTasks}
+        restart={shouldRestart}
+        setMutationLoading={setMutationLoading}
+        setButtonDisabled={setButtonDisabled}
+        buttonDisabled={buttonDisabled}
+      />
+      <>
+        {childPatches && (
+          <div data-cy="select-downstream">
+            <HR />
+            <Accordion
+              title={<StyledTitle> Select Downstream Tasks</StyledTitle>}
+              contents={childPatches.map(({ id: childPatchId, projectID }) => (
+                <Accordion
+                  key={childPatchId}
+                  title={projectID}
+                  contents={
+                    <PatchRestartTasks
+                      patchId={childPatchId}
+                      visible={visible}
+                      onOk={onOk}
+                      onCancel={onCancel}
+                      refetchQueries={refetchQueries}
+                      setShouldAbortInProgressTasks={
+                        setShouldAbortInProgressTasks
+                      }
+                      shouldAbortInProgressTasks={shouldAbortInProgressTasks}
+                      restart={shouldRestart}
+                      setMutationLoading={setMutationLoading}
+                      setButtonDisabled={setButtonDisabled}
+                      buttonDisabled={buttonDisabled}
+                    />
+                  }
+                />
+              ))}
+            />
+          </div>
+        )}
+        <PatchRestartFooter
+          setShouldAbortInProgressTasks={setShouldAbortInProgressTasks}
+          shouldAbortInProgressTasks={shouldAbortInProgressTasks}
+        />
+      </>
+    </>
   );
+};
+
+interface PatchRestartTasksProps {
+  restart: number;
+  visible: boolean;
+  onOk: () => void;
+  onCancel: () => void;
+  patchId?: string;
+  refetchQueries: string[];
+  setShouldAbortInProgressTasks: React.Dispatch<React.SetStateAction<boolean>>;
+  shouldAbortInProgressTasks: boolean;
+  setMutationLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setButtonDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+  buttonDisabled: boolean;
+}
+export const PatchRestartTasks: React.FC<PatchRestartTasksProps> = ({
```

Comment:
we should move this export into a separate file where the filename matches the export name